os/board/rtl8720e/src/component/os/tizenrt/rtk_blemgr.c

```diff
@@ -0,0 +1,439 @@
+{
+	printf("[%s : %d]\n", __FILE__, __LINE__);
+	trble_result_e ret = TRBLE_INVALID_ARGS;
+	// temporary remove
+	// if (!rltk_wlan_running(0)) {
+	// 	TRBLE_TEST_ERR("[TRBLE] WiFi is off. Please turn on WiFi first.\n");
+	// 	return TRBLE_FAIL;
+	// }
+#ifdef CONFIG_AMEBALITE_BLE_SCATTERNET
+	if (client != NULL && server != NULL) {
+		ret = rtw_ble_combo_init(client, server);
+	}
+#elif defined(CONFIG_AMEBALITE_BLE_CENTRAL)
+	if (client != NULL) {
+		ret = rtw_ble_client_init(client);
+	}
+#elif defined(CONFIG_AMEBALITE_BLE_PERIPHERAL)
+	if (server != NULL) {
+		ret = rtw_ble_server_init(server);
+	}
+#endif
+	/* Set WiFi time slot */
+	// temporary remove
+	// (void)rltk_coex_set_ble_scan_duty(WIFI_TIME_SLOT);
+	if (ret == TRBLE_SUCCESS) {
+		ret = rtw_ble_server_get_mac_address(dev->hwaddr);
+
+		if (ret != TRBLE_SUCCESS) {
+			TRBLE_TEST_ERR("[TRBLE] Fail to get BLE mac\n");
+		} else {
+			_reverse_mac(dev->hwaddr, NULL);
+		}
+	}
+	return ret;
+}
+
+trble_result_e trble_netmgr_deinit(struct bledev *dev)
+{
+	trble_result_e ret = TRBLE_FAIL;
+#ifdef CONFIG_AMEBALITE_BLE_SCATTERNET
+	ret = rtw_ble_combo_deinit();
+#elif defined(CONFIG_AMEBALITE_BLE_CENTRAL)
+	ret = rtw_ble_client_deinit();
+#elif defined(CONFIG_AMEBALITE_BLE_PERIPHERAL)
+	ret = rtw_ble_server_deinit();
+#endif
+	if (ret == TRBLE_SUCCESS) {
+		memset(dev->hwaddr, 0, TRBLE_BD_ADDR_MAX_LEN);
+	}
+	return ret;
+}
+
+trble_result_e trble_netmgr_get_mac_addr(struct bledev *dev, uint8_t mac[TRBLE_BD_ADDR_MAX_LEN])
+{
+	memcpy(mac, dev->hwaddr, TRBLE_BD_ADDR_MAX_LEN);
+	return TRBLE_SUCCESS;
+}
+
+trble_result_e trble_netmgr_get_bonded_device(struct bledev *dev, trble_bonded_device_list_s *device_list, uint16_t *device_count)
+{
+	int i;
+	trble_result_e ret = rtw_ble_server_get_bonded_device(device_list, device_count);
+	
+	if (ret == TRBLE_SUCCESS) {
+		for (i = 0; i < *device_count; i++) {
+			_reverse_mac(device_list[i].bd_addr.mac, NULL);
+		}
+	}
+
+	return ret;
+}
+
+trble_result_e trble_netmgr_delete_bond(struct bledev *dev, trble_addr *addr)
+{
+	trble_addr t_addr[1] = { 0, };
+	t_addr->type = addr->type;
+	_reverse_mac(addr->mac, t_addr->mac);
+	return rtw_ble_client_delete_bond(t_addr);
+}
+
+trble_result_e trble_netmgr_delete_bond_all(struct bledev *dev)
+{
+	return rtw_ble_client_delete_bond_all();
+}
+
+trble_result_e trble_netmgr_conn_is_active(struct bledev *dev, trble_conn_handle con_handle, bool *is_active)
+{
+	*is_active = rtw_ble_server_conn_is_active(con_handle);
+	return TRBLE_SUCCESS;
+}
+
+trble_result_e trble_netmgr_conn_is_any_active(struct bledev *dev, bool *is_active)
+{
+	*is_active = rtw_ble_server_conn_is_any_active();
+	return TRBLE_SUCCESS;
+}
+
+trble_result_e trble_netmgr_ioctl(struct bledev *dev, trble_msg_s *msg)
+{
+	trble_result_e ret = TRBLE_UNSUPPORTED;
+	
+	if (msg->cmd == TRBLE_MSG_GET_VERSION) {
+		uint8_t *version = (uint8_t *)msg->data;
+		// temporary remove
+		// ret = rtw_ble_get_version(version);
+	}
+	return ret;

```

Comment:
It always returns `TRBLE_UNSUPPORTED`.

Reply:
At the moment rtw_ble_get_version( ) is not supported, therefore it will always return TRBLE_UNSUPPORTED.